tools/dotted-pascal-case-filter/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+use heck::ToPascalCase;
+
+wit_bindgen_rust::export!("./custom-filter.wit");
+
+struct CustomFilter;
```

Comment:
Should this be named specific to the type of CustomFilter it is? ie `DottedPascalCaseCustomFilter`. It's wordy but if we expect more to come, it may be helpful.

Reply:
THAT WOULD BE SO NICE.

Unfortunately, the way the `wit-bindgen` macros work is that the emitted code refers to `super::<PascalisedWITFileName>`.  So the struct implementing the trait _must_ match the name of the WIT file.  If you call it something meaningful, you get enigmatic compile errors about `cannot find type 'CustomFilter' in module 'super'`.  (My first test suites did exactly this and I ended up having to cargo expand them to understand why they weren't working.)  It's a bit half-arsed.